core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -244,6 +244,25 @@ object ParserGen {
     val fn: A => F[B]
   }
 
+  def selected(ga: Gen[GenT[Parser]]): Gen[GenT[Parser]] =
+    Gen.zip(ga, pures).flatMap { case (parser, genRes) =>
+      val gpab = Gen.either(parser.fa, genRes.fa.map(Parser.pure)).map {
```

Comment:
We need a `B` that can be the right of `gpab` and also the return type of the function in `gfn`.  This compiles, but the `Parser.pure` feels shoddy.  I'll try Level 2 Type Tetris later.

Reply:
Yeah my claim is with a pure here the laws won't pass without more changes.

Reply:
Okay, Type Tetris Two looks more like the tailRecM case, and got us some failing tests.  I think now we can see if your other suggestions green it back up.